src/components/interaction-designer/block-editors/ChoicesBuilder.vue

```diff
@@ -0,0 +1,143 @@
+        class="choices-builder-item"
+        :label="(choiceIndex + 1).toString()"
+        :rows="1"
+        :placeholder="'Enter choice...'"
+        :resource-id="resourceId"
+        :resource-variant="resourcesByUuid[resourceId].values[0]"
+        :mode="SupportedMode.TEXT"
+        @afterResourceVariantChanged="handleExistingResourceVariantChangedFor(
+          resourceId,
+          choiceIndex,
+          resourcesByUuid[resourceId])" />
+    </template>
+
+    <!--Show empty choice-->
+    <resource-variant-text-editor
+      v-if="draftResource"
+      ref="draftChoice"
+      :key="draftResource.uuid"
+      class="choices-builder-item"
+      :label="(choiceKeys.length + 1).toString()"
+      :rows="1"
+      :placeholder="'Enter choice...'"
+      :resource-id="draftResource.uuid"
+      :resource-variant="findOrGenerateStubbedVariantOn(
+        draftResource,
+        {language_id: activeFlow.languages[0].id,
+         content_type: SupportedContentType.TEXT,
+         modes: [SupportedMode.TEXT]})"
+      :mode="SupportedMode.TEXT"
+      @beforeResourceVariantChanged="addDraftResourceToChoices"
+      @afterResourceVariantChanged="rewriteChoiceKeyFor({resourceId: draftResource.uuid, blockId: block.uuid})" />
+  </div>
+</template>
+
+<script lang="ts">
+import {get, isEmpty, last} from 'lodash'
+import {findOrGenerateStubbedVariantOn} from '@/store/flow/resource'
+import {Component, Prop} from 'vue-property-decorator'
+import {mixins} from 'vue-class-component'
+import Lang from '@/lib/filters/lang'
+import {IBlock, IBlockExit, IFlow, IResource, SupportedContentType, SupportedMode} from '@floip/flow-runner'
+import {namespace} from 'vuex-class'
+import ResourceVariantTextEditor from '@/components/interaction-designer/resource-editors/ResourceVariantTextEditor.vue'
+import {ISelectOneResponseBlock} from '@floip/flow-runner/src/model/block/ISelectOneResponseBlock'
+import SelectOneStore, {BLOCK_TYPE} from '@/store/flow/block-types/MobilePrimitives_SelectOneResponseBlockStore'
+import {IdGeneratorUuidV4} from '@floip/flow-runner/dist/domain/IdGeneratorUuidV4'
+import Vue from 'vue'
+
+const flowVuexNamespace = namespace('flow')
+const blockVuexNamespace = namespace(`flow/${BLOCK_TYPE}`)
+
+@Component({
+  components: {
+    ResourceVariantTextEditor,
+
+  },
+})
+class ChoicesBuilder extends mixins(Lang) {
+  @Prop() readonly block!: ISelectOneResponseBlock
+
+  draftResource: IResource | null = null
+
+  SupportedContentType = SupportedContentType
+  SupportedMode = SupportedMode
+  findOrGenerateStubbedVariantOn = findOrGenerateStubbedVariantOn
+
+  get choiceKeys(): string[] {
+    // todo: my best guess for ordering is resources ordering? :/
+    return Object.keys(this.block.config.choices)
+  }
+
+  created(): void {
+    this.generateDraftResource()
+  }
+
+  async generateDraftResource(): Promise<void> {
+    this.draftResource = await this.resource_createWith({ // todo: abstract as createResourceForVanillaChoice()
```

Comment:
Same in this TODO
is it possible to track it in a ticket if it is still relevant ?